File: Python/Software Development Module - Python/Week11/AssessmentApp.py
"""
AssessmentApp.py
Billy Cussen
06/10/2020
"""

#Objects
import gc
import re
import Assessment 
from Assessment import myAssessment
import Test 
from Test import myTest
import Project
from Project import myProject

#Variables
repeat = "y"
inputCheck = True

while repeat == "y":

    #Input - Assessment Type
    while inputCheck:
        try:
            typeOfAssessment = str(input("Please enter Assessment Type (\"Test\" or \"Project\"): "))
            typeOfAssessment = typeOfAssessment.lower()
        except:
            print("You need to enter your appointment type!")
        else:
            if(typeOfAssessment != "test" and typeOfAssessment != "project"):
                print("You need to enter \"Test\" or \"Project\"!")
            else:
                inputCheck = False

    #Input - Weighting
    while inputCheck == False:
        try:
            weighting = int(input("Please enter your Weighting of this "+str(typeOfAssessment)+" (e.g. 60): "))
        except:
            print("You need to enter a number to Denote the Weighting of this "+str(typeOfAssessment)+"!")
        else:
            if(weighting <= 0 or weighting > 100):
                print("You need to enter a number between 1 and 100!")
            else:
                inputCheck = True

    #Input - Name
    while inputCheck:
        try:
            name  = str(input("Please enter the Name of this "+str(typeOfAssessment)+" (e.g. \"OOP Programming\"): "))
            name  = name.lower()
        except:
            print("You need to enter a valid name!")
        else:
            if(len(name) <= 0):
                print("You need to enter a valid name for this "+str(typeOfAssessment)+"!")
            else:
                inputCheck = False


    if(typeOfAssessment == "test"):
        while inputCheck == False:
            try:
                questions = int(input("Please enter the number of question in this "+str(typeOfAssessment)+" (e.g. 60): "))
            except:
                print("You need to enter the NUMBER of questions in this "+str(typeOfAssessment)+"!")
            else:
                if(questions <= 0):
                    print("You need to enter a valid number of questions!")
                else:
                    inputCheck = True

        while inputCheck:
            try:
                duration  = int(input("Please enter the Duration of this "+str(typeOfAssessment)+" in minutes (e.g. 60): "))
            except:
                print("You need to enter a valid number!")
            else:
                if(duration <= 0):
                    print("You need to enter a valid Duration for this "+str(typeOfAssessment)+"!")
                else:
                    inputCheck = False

        #Process & Output
        try:
            myTestApp = myTest(weighting, name, typeOfAssessment, questions, duration)
            print(myTestApp.toString())
        except Exception as e:
            print(str(e))
        finally:
            myAssessmentApp = None
            gc.collect()
    else:
        while inputCheck == False:
            try:
                dateTemp = str(input("Enter the Deadline Date of this "+str(typeOfAssessment)+" (In the format DD/MM/YYYY): "))
                checkDate = re.search("[0-9]{2}/[0-9]{2}/[0-9]{4}", dateTemp)
                if checkDate == False:
                    raise Exception()
            except:
                print("You need to enter a valid Deadline Date for this "+str(typeOfAssessment)+"!")
            else:
                if(len(dateTemp) < 8):
                    print("You need to enter a valid Date for the Deadline!")
                else:
                    inputCheck = True

        #Process & Output
        try:
            myAssessmentApp = myProject(weighting, name, typeOfAssessment, dateTemp)
            print(myAssessmentApp.toString())
        except Exception as e:
            print(str(e))
        finally:
            myAssessmentApp = None
            gc.collect()


    #Input - Repeat?
    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter \"Y\" to continue or \"N\" to quit: "))
        repeat = repeat.lower()
        if repeat != "y" and repeat !="n":
            print("You need to enter \"y\" or \"n\"")
    
print("Thanks for using my App!")

File: Java/Software Development Module - Java/Week11/AssessmentApp.java
/*
AssessmentApp.java
Billy Cussen
21/07/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class AssessmentApp {
    
    public static void main(String[]args){

        //Variables
        int weighting = 0, questions = 0, duration = 0;
        String name = new String(), type = new String();
        Date deadline = new Date();
        String output;
        boolean inputCheck;

        //Objects
        Scanner keyboard;
        Assessment myAssessment;
        Test myTest;
        Project myProject;

        do{
            keyboard = new Scanner(System.in);
            
            try{
                System.out.print("Please enter Assessment Type (Test or Project): ");
                type = keyboard.nextLine();

                if(!(type.equalsIgnoreCase("Test")||type.equalsIgnoreCase("Project"))){
                    throw new InputMismatchException();
                }

            }catch(InputMismatchException e){
                System.out.println("Please enter either Test or Project!");
            }finally{
                keyboard = null;
            }

        }while(!(type.equalsIgnoreCase("Test")||type.equalsIgnoreCase("Project")));

        do{
            keyboard = new Scanner(System.in);
            inputCheck = true;

            try{
                System.out.print("Enter the weighting of this "+type+" (e.g. 60): ");
                weighting = keyboard.nextInt();

                if(weighting <= 0 || weighting > 100){
                    throw new InputMismatchException();
                }

            }catch(InputMismatchException e){
                System.out.println("Please enter a number between 1 and 100 to denote weighting");
                inputCheck = false;
            }finally{
                keyboard = null;
            }

        }while(!inputCheck);           

        do{
            keyboard = new Scanner(System.in);
            inputCheck = true;

            try{
                System.out.print("Enter the Name of this "+type+" (e.g. OOP Project): ");
                name = keyboard.nextLine();

                if(name.length() <= 0){
                    throw new InputMismatchException();
                }

            }catch(InputMismatchException e){
                System.out.println("Please enter a valid name for this "+type);
                inputCheck = false;
            }finally{
                keyboard = null;
            }

        }while(!inputCheck);       

        if(type.equalsIgnoreCase("Test")){

            do{
                keyboard = new Scanner(System.in);
                inputCheck = true;
    
                try{
                    System.out.print("Enter the number of questions in this "+type+": ");
                    questions = keyboard.nextInt();
    
                    if(questions <= 0){
                        throw new InputMismatchException();
                    }
    
                }catch(InputMismatchException e){
                    System.out.println("Please enter a valid number of questions for this "+type);
                    inputCheck = false;
                }finally{
                    keyboard = null;
                }
    
            }while(!inputCheck);

            do{
                keyboard = new Scanner(System.in);
                inputCheck = true;
    
                try{
                    System.out.print("Enter the Duration of this "+type+" in minutes (e.g. 60): ");
                    duration = keyboard.nextInt();
    
                    if(duration <= 0){
                        throw new InputMismatchException();
                    }
    
                }catch(InputMismatchException e){
                    System.out.println("Please enter a valid duration for this "+type);
                    inputCheck = false;
                }finally{
                    keyboard = null;
                }
    
            }while(!inputCheck);

            myTest = new Test(weighting,name,type,questions,duration);
            System.out.println(myTest.toString());
            myTest = null;

        } else{

            do{
                keyboard = new Scanner(System.in);
                inputCheck = true;
    
                try{
                    System.out.print("Enter the Deadline Date of this "+type+" (In the format DD/MM/YYYY): ");
                    String dateTemp = keyboard.nextLine();
                    deadline = new SimpleDateFormat("dd/MM/yyyy").parse(dateTemp);
                    
                    if(dateTemp.length() < 8){
                        throw new InputMismatchException();
                    }

                }catch(InputMismatchException | ParseException e){
                    System.out.println("Please enter a valid DeadlineDate for this "+type);
                    inputCheck = false;
                }finally{
                    keyboard = null;
                }
    
            }while(!inputCheck);

            myProject = new Project(weighting,name,type,deadline);
            System.out.println(myProject.toString());
            myProject = null;

        }


    }

}